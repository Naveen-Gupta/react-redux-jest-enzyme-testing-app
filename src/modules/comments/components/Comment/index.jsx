import React from 'react';
import PropTypes from 'prop-types';
import Card from '@tds/core-card';
import Box from '@tds/core-box';
import Heading from '@tds/core-heading';
import Text from '@tds/core-text';

const Comment = (props) => {
    const { commentDetails } = props;

    if(commentDetails){
        return (
                <Card data-test="wrapperComment" variant="branded" onClick={() => props.history.push(`/edit/${commentDetails.id}`)}>
                    <Box between={3} vertical={2}>
                        <Heading data-test="titleComment" level="h3">{commentDetails.title}</Heading>
                        <Text data-test="textComment">
                            {commentDetails.comment}
                        </Text>
                        <Text data-test="nameComment">{`By: ${commentDetails.name}`}</Text>
                    </Box>
                </Card>
        );
    } else {
        return <div data-test="wrapperComment">{null}</div>;
    }
    
   
}

Comment.propTypes = {
    history: PropTypes.array,
    commentDetails: PropTypes.object
}

Comment.defaultProps = {
    commentDetails: null,
    history:[]
}

export default Comment;