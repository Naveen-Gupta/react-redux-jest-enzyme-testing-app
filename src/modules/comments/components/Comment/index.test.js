import React from 'react';
import { shallow, mount } from 'enzyme';

import { findByTestAtrr } from '../../../../utils';
import Comment from './index.jsx';

const setUp = (props = {}) => {
	const component = shallow(<Comment {...props} />);
	return component;
};

describe('Comment Component', () => {
	let component;
	beforeEach(() => {
		component = setUp();
	});

	test('Should render without errors', () => {
		const wrapper = findByTestAtrr(component, 'wrapperComment');
		expect(wrapper.length).toBe(1);
	});

	describe('Have props', () => {
		let wrapper;
		const props = {
			commentDetails: {
				id: 1,
				title: 'Test title',
				name: 'Test name',
				comment: 'Test comment',
			},
			history: [],
    };
    
		beforeEach(() => {
			wrapper = setUp({...props});
		});

		test('Should render without errors', () => {
			const component = findByTestAtrr(wrapper, 'wrapperComment');
			expect(component.length).toBe(1);
    });
    
    test('Should render a comment with title "Test title"', () => {
      const titleComment = findByTestAtrr(wrapper, 'titleComment');
      expect(titleComment.containsMatchingElement("Test title")).toBeTruthy();
    });
    
    test('Should render a comment with name "Test name"', () => {
      const nameComment = findByTestAtrr(wrapper, 'nameComment');
      expect(nameComment.containsMatchingElement("By: Test name")).toBeTruthy();
    });

    test('Should render a comment with text "Test comment"', () => {
      const textComment = findByTestAtrr(wrapper, 'textComment');
      expect(textComment.containsMatchingElement("Test comment")).toBeTruthy();
    });

		test('Checking snapshot with providing props', () => {
			const wrap = mount(<Comment {...props} />);
			expect(wrap).toMatchSnapshot();
    });
    
  });
  
  describe('Does not have props', () => {
		let wrapper;
    
		beforeEach(() => {
			wrapper = setUp();
		});

		test('Should render without errors', () => {
			const component = findByTestAtrr(wrapper, 'wrapperComment');
			expect(component.length).toBe(1);
		});

		test('Checking snapshot without providing props', () => {
			const wrap = mount(<Comment />);
			expect(wrap).toMatchSnapshot();
		});
	});
});
