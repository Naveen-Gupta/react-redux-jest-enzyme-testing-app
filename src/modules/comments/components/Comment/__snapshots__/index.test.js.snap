// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component Does'nt have props Checking snapshot without providing props 1`] = `
<Comment
  commentDetails={null}
  history={Array []}
>
  <div
    data-test="wrapperComment"
  />
</Comment>
`;

exports[`Comment Component Have props Checking snapshot with providing props 1`] = `
<Comment
  commentDetails={
    Object {
      "comment": "Test comment",
      "id": 1,
      "name": "Test name",
      "title": "Test title",
    }
  }
  history={Array []}
>
  <Card
    data-test="wrapperComment"
    fullHeight={false}
    key="1"
    onClick={[Function]}
    spacing="default"
    variant="branded"
  >
    <Styled(Component)
      data-test="wrapperComment"
      fullHeight={false}
      horizontal={4}
      onClick={[Function]}
      variant="branded"
      vertical={5}
    >
      <Component
        className="sc-AxirZ bvZuyl"
        data-test="wrapperComment"
        fullHeight={false}
        horizontal={4}
        onClick={[Function]}
        variant="branded"
        vertical={5}
      >
        <Box
          className="sc-AxirZ bvZuyl"
          data-test="wrapperComment"
          horizontal={4}
          inline={false}
          onClick={[Function]}
          tag="div"
          variant="branded"
          vertical={5}
        >
          <styled.div
            className="sc-AxirZ bvZuyl"
            data-test="wrapperComment"
            horizontal={4}
            inline={false}
            onClick={[Function]}
            tag="div"
            variant="branded"
            vertical={5}
          >
            <div
              className="sc-AxjAm fLAVMH sc-AxirZ bvZuyl sc-AxirZ bvZuyl"
              data-test="wrapperComment"
              onClick={[Function]}
            >
              <Box
                between={3}
                inline={false}
                tag="div"
                vertical={2}
              >
                <styled.div
                  between={3}
                  inline={false}
                  tag="div"
                  vertical={2}
                >
                  <div
                    className="sc-AxjAm gyKeVc"
                  >
                    <Heading
                      data-test="titleComment"
                      invert={false}
                      level="h3"
                    >
                      <Heading__StyledHeading
                        as="h3"
                        data-test="titleComment"
                        data-testid="heading"
                        invert={false}
                        level="h3"
                      >
                        <h3
                          className="Heading__StyledHeading-mpz92r-0 erviEy"
                          data-test="titleComment"
                          data-testid="heading"
                        >
                          Test title
                        </h3>
                      </Heading__StyledHeading>
                    </Heading>
                    <Text
                      block={false}
                      bold={false}
                      data-test="textComment"
                      invert={false}
                      size="base"
                    >
                      <Text__StyledText
                        block={false}
                        bold={false}
                        data-test="textComment"
                        invert={false}
                        size="base"
                      >
                        <span
                          className="Text__StyledText-sc-1m0rr67-0 dEioGg"
                          data-test="textComment"
                          size="base"
                        >
                          Test comment
                        </span>
                      </Text__StyledText>
                    </Text>
                    <Text
                      block={false}
                      bold={false}
                      data-test="nameComment"
                      invert={false}
                      size="base"
                    >
                      <Text__StyledText
                        block={false}
                        bold={false}
                        data-test="nameComment"
                        invert={false}
                        size="base"
                      >
                        <span
                          className="Text__StyledText-sc-1m0rr67-0 dEioGg"
                          data-test="nameComment"
                          size="base"
                        >
                          By: Test name
                        </span>
                      </Text__StyledText>
                    </Text>
                  </div>
                </styled.div>
              </Box>
            </div>
          </styled.div>
        </Box>
      </Component>
    </Styled(Component)>
  </Card>
</Comment>
`;
